src/basic/unit-name.c

```diff
@@ -581,6 +629,9 @@ int unit_name_to_path(const char *name, char **ret) {
         if (r < 0)
                 return r;
 
+        if (strstr(prefix, "..."))
+                return -ENAMETOOLONG;
+
         return unit_name_path_unescape(prefix, ret);
 }
```

Comment:
the various callers of unit_name_to_path() should probaby be updated to handle this error nicely in some way.

Reply:
I've checked this previously and I think that they should be fine but I have double check on that.